File: test-plugins/org.yakindu.sct.simulation.core.sexec.test/test-gen/org/yakindu/sct/simulation/core/sexec/test/DeepHistoryTest.java
/**
 * Copyright (c) 2013 committers of YAKINDU and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     committers of YAKINDU - initial API and implementation
 */
package org.yakindu.sct.simulation.core.sexec.test;
import org.eclipse.xtext.junit4.InjectWith;
import org.eclipse.xtext.junit4.XtextRunner;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.yakindu.sct.model.sexec.ExecutionFlow;
import org.yakindu.sct.model.sexec.interpreter.test.util.AbstractExecutionFlowTest;
import org.yakindu.sct.model.sexec.interpreter.test.util.SExecInjectionProvider;
import org.yakindu.sct.test.models.SCTUnitTestModels;
import com.google.inject.Inject;
import static org.junit.Assert.assertTrue;
/**
 *  Unit TestCase for DeepHistory
 */
@SuppressWarnings("all")
@RunWith(XtextRunner.class)
@InjectWith(SExecInjectionProvider.class)
public class DeepHistoryTest extends AbstractExecutionFlowTest {
	@Inject
	private SCTUnitTestModels models;
	@Before
	public void setup() throws Exception {
		ExecutionFlow flow = models
				.loadExecutionFlowFromResource("DeepHistory.sct");
		initInterpreter(flow);
	}
	@Test
	public void deepHistoryTest() throws Exception {
		interpreter.enter();
		raiseEvent("event1");
		interpreter.runCycle();
		raiseEvent("event3");
		interpreter.runCycle();
		raiseEvent("event5");
		interpreter.runCycle();
		raiseEvent("event7");
		interpreter.runCycle();
		assertTrue(!isActive("State1"));
		assertTrue(isActive("State9"));
		raiseEvent("event2");
		interpreter.runCycle();
		assertTrue(isActive("State1"));
		assertTrue(!isActive("State9"));
		raiseEvent("event1");
		interpreter.runCycle();
		assertTrue(!isActive("State1"));
		assertTrue(isActive("State9"));
	}
}

File: test-plugins/org.yakindu.sct.generator.c.test/gtests/DeepHistory/DeepHistoryTest.cc
/**
* Copyright (c) 2017 committers of YAKINDU and others.
* All rights reserved. This program and the accompanying materials
* are made available under the terms of the Eclipse Public License v1.0
* which accompanies this distribution, and is available at
* http://www.eclipse.org/legal/epl-v10.html
*
* Contributors:
*     committers of YAKINDU - initial API and implementation
*/
#include "gtest/gtest.h"
#include "DeepHistory.h"

static DeepHistory statechart;

class StatemachineTest : public ::testing::Test{
	protected:
	virtual void SetUp() {
		deepHistory_init(&statechart);
	}
};


TEST_F(StatemachineTest, deepHistoryTest) {					
	deepHistory_enter(&statechart);
	deepHistoryIface_raise_event1(&statechart);
	deepHistory_runCycle(&statechart);
	deepHistoryIface_raise_event3(&statechart);
	deepHistory_runCycle(&statechart);
	deepHistoryIface_raise_event5(&statechart);
	deepHistory_runCycle(&statechart);
	deepHistoryIface_raise_event7(&statechart);
	deepHistory_runCycle(&statechart);
	EXPECT_TRUE(!deepHistory_isStateActive(&statechart, DeepHistory_mainRegion_State1));
	EXPECT_TRUE(deepHistory_isStateActive(&statechart, DeepHistory_mainRegion_State2__region0_State4__region0_State7__region0_State9));
	deepHistoryIface_raise_event2(&statechart);
	deepHistory_runCycle(&statechart);
	EXPECT_TRUE(deepHistory_isStateActive(&statechart, DeepHistory_mainRegion_State1));
	EXPECT_TRUE(!deepHistory_isStateActive(&statechart, DeepHistory_mainRegion_State2__region0_State4__region0_State7__region0_State9));
	deepHistoryIface_raise_event1(&statechart);
	deepHistory_runCycle(&statechart);
	EXPECT_TRUE(!deepHistory_isStateActive(&statechart, DeepHistory_mainRegion_State1));
	EXPECT_TRUE(deepHistory_isStateActive(&statechart, DeepHistory_mainRegion_State2__region0_State4__region0_State7__region0_State9));
}


